src/cascadia/TerminalSettingsModel/ColorScheme.idl

```diff
@@ -11,7 +11,10 @@ namespace Microsoft.Terminal.Settings.Model
         Windows.UI.Color SelectionBackground;
         Windows.UI.Color CursorColor;
 
-        Windows.UI.Color[] Table { get; };
+        // winrt::com_arrays prevent data binding.
+        // Instead of representing Table as a property,
+        // we expose the getter as a function.
+        Windows.UI.Color[] Table();

```

Comment:
wait but why tho? Are we using `Table()` anywhere?

Reply:
So, the funny thing is, `Table { get; };` and `Table();` in the IDL are both called with `Table()` in C++. So this doesn't affect anybody calling it haha.

As for why this is happening, the compiler is asserting that `winrt::com_array<winrt::Windows::UI::Color>` must be a WinRT type so that it can be wrapped in an `IReference`. Why a `winrt::com_array` isn't a WinRT type, idk. So changing it to a function is just easier haha

Reply:
Wait but why is the compiler asserting this now, and not before?

Reply:
Ah, because of this DataTemplate in Profiles.xaml:
```xml
<DataTemplate x:DataType="model:ColorScheme">
   <TextBlock Text="{x:Bind Name, Mode=OneWay}"/>
</DataTemplate>
```

Even thought we're _only_ binding `Name`, XAML tries to make _every property_ bindable. Since Table was a property, we'd hit that error :/